# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLSync.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLSync.h
--- a/dom/canvas/WebGLSync.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLSync.h	Wed Jan 08 22:19:16 2020 +0000
@@ -6,8 +6,6 @@
 #ifndef WEBGL_SYNC_H_
 #define WEBGL_SYNC_H_
 
-#include "mozilla/LinkedList.h"
-#include "nsWrapperCache.h"
 #include "WebGLObjectModel.h"
 
 namespace mozilla {
@@ -15,11 +13,12 @@
 class AvailabilityRunnable;
 }  // namespace webgl
 
-class WebGLSync final : public WebGLRefCountedObject<WebGLSync>,
-                        public LinkedListElement<WebGLSync> {
+class WebGLSync final : public WebGLContextBoundObject {
   friend class WebGL2Context;
   friend class webgl::AvailabilityRunnable;
 
+  MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(WebGLSync, override)
+
   const GLsync mGLName;
   const uint64_t mFenceId;
   bool mCanBeAvailable = false;
@@ -27,14 +26,10 @@
  public:
   WebGLSync(WebGLContext* webgl, GLenum condition, GLbitfield flags);
 
-  void Delete();
-
-  NS_INLINE_DECL_REFCOUNTING(WebGLSync)
-
   void MarkSignaled() const;
 
  private:
-  ~WebGLSync();
+  ~WebGLSync() override;
 };
 
 }  // namespace mozilla